### DISCUSSION TITLE
Operators that change the data type: widen applicability and improve usability

### DISCUSSION THREAD

**Discussion by amattioc** [sdmx-twg/vtl#409] [General]
The behaviour of operators that return a data type different from the operands is described in the RM (row 1373). Anyway, there are many other operators that behave this way (at least under certain conditions) and with the current rules their usability in real world scenarios is quite low. An example can be found in the discussion related to issue #404 and in issues #407 and #406. The counter-intuitiveness of the current rules is proven by the many wrong examples for those operators in the RM v2.0.

**Comment by linardian** [comment]
I think we can close this issue (e and others correlated), given the discussion (and the proposal) not to considerate "changing of data type" the "generalisation" from a sub-type to the corrispondent type. If so please consider to close this issue and #404, #406 and #407. @amattioc  is ok for you ?

**Reply by amattioc** [reply]
I agree
